src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -126,12 +126,8 @@ public void LogError(string message, params object[] args)
 
         public void LogDebug(string message, params object[] args)
         {
-            if (Verbosity == LoggerVerbosity.Debug)
-            {
-                Write(MessageType.Debug, message, args);
-            }
+            Write(MessageType.Debug, message, args);
```

Comment:
Previously, the verbosity was checked here and the message discarded immediately if the verbosity wasn't debug.

Now, we're attempting to write all messages, and doing the filtering in the `Write` method. That way, debug messages will be recorded when output logging is suspended, and then filtered out based on the verbosity in force when `ResumeOutput` is called.